src/megatron/energon/savable_loader.py

```diff
@@ -546,21 +491,9 @@ def get_checkpoint(self, last_sample_indexes: List[int]) -> SavableDatasetCheckp
                 )
         raise ValueError("No checkpoint found")
 
-    def merge_checkpoints(
-        self, checkpoints: List[SavableDatasetCheckpoint]
-    ) -> SavableDatasetMergedCheckpoint:
-        """Merges saved checkpoints from all worker processes."""
-        assert torch.utils.data.get_worker_info() is None, "Cannot merge in worker process"
-        assert all(isinstance(c, SavableDatasetCheckpoint) for c in checkpoints)
-        return SavableDatasetMergedCheckpoint(
-            dataset_state=self.dataset.merge_states([c.dataset_state for c in checkpoints]),
-            state=self._merge_states([c.state for c in checkpoints]),
-            offset=[c.offset for c in checkpoints],
-        )
-
     def restore_checkpoint(
         self,
-        state: Optional[SavableDatasetMergedCheckpoint],
+        state_list: Optional[List[SavableDatasetCheckpoint]],
```

Comment:
Docs are not adapted. Also, why `state` to `state_list`? I think `list` does not help to understand that better (and typing already makes clear, it's a list :D ). Rename e.g. to `workers_state`.